### DISCUSSION TITLE
تفکیک لینک سابسکریپشن و پنل در مرزبان

### DISCUSSION THREAD

**Discussion by mdjvd** [Gozargah/Marzban#684] [Ideas]
سلام بر شما. ممکنه نیاز باشه دامنه پنل و دامنه لینک سابسکریپشن رو از همدیگه جدا کنید. (حالا به هر دلیلی)
برای این کار چند روش وجود داره:

۱- میتونید بعد از گرفتن یک سرتیفیکیت برای دامنه اول و فعالسازی SSL در مرزبان، فقط تیک پروکسی دامنه دوم رو فعال کنید و برای پنل یا لینک سابسکریپشن در نظر بگیریدش.

۲- اگر میخواید از یک دامنه یکسان استفاده کنید ولی سابدامنه های متفاوت برای پنل و لینک سابسکریپشن بذارید، میتونید برای دامنه‌تون یک سرتیفیکیت `wildcard` بگیرید. (پیشنهاد نمیشه. بهتره از دامنه‌های کاملا متفاوت استفاده کنید.)

**۳- یک سرتیفیکیت برای دو دامنه بگیرید. توی این آموزش ما این کارو انجام میدیم.**

 _خب برای گرفتن یک سرتیفیکیت برای دو دامنه، ما با `acme.sh` انجامش میدیم ولی شما میتونید از `certbot` هم استفاده کنید. (البته با دستورهای خودش)_ 

- اول پیش‌نیازها رو نصب کنید:
```
apt-get install -y curl cron socat
```

- با دستور زیر `acme.sh` رو نصب کنید: 
_(به جای `MYEMAIL@gmail.com` ایمیل خودتون رو قرار بدید)_
```
curl https://get.acme.sh | sh -s email=MYEMAIL@gmail.com
```

توجه: در مرحله بعد نیازه که پورت 80 روی سرور شما در دسترس و در فایروال (اگر فعاله) باز شده باشه. ممکنه روی سرور از قبل nginx یا apache نصب شده باشه و پورت 80 توسطشون مشغول شده باشه، که میتونید با یکی از این دو دستور سرویس این دو رو موقتا متوقف کنید و بعد از دریافت سرتیفیکیت در صورت نیاز مجددا فعالشون کنید:
`systemctl stop nginx`
`systemctl stop apache2.service`

- با دستور زیر سرتیفیکیت رو دریافت کنید:
_(به جای `PANEL.DOMAIN1.COM` و `SUB.DOMAIN2.COM` سابدامنه‌های خودتون رو قرار بدید)_
```
~/.acme.sh/acme.sh --set-default-ca --server letsencrypt --issue --force --standalone \
-d PANEL.DOMAIN1.COM -d SUB.DOMAIN2.COM
```

- بعدش سرتیفیکیت گرفته شده رو در فولدر مرزبان نصب کنید:
_(دقت کنید که توی این دستور باید فقط اولین دامنه در دستور قبلی رو قرار بدید. در نمونه بالا یعنی دامنه `PANEL.DOMAIN1.COM`)_

```
~/.acme.sh/acme.sh --installcert -d PANEL.DOMAIN1.COM \
--key-file /var/lib/marzban/certs/key.pem \
--fullchain-file /var/lib/marzban/certs/fullchain.pem
```

- حالا وارد فایل `env` بشید:
```
nano /opt/marzban/.env
```

![15f7cb361d913ba993cf7](https://github.com/Gozargah/Marzban/assets/116950557/42c7ff2f-b931-44e9-b562-0bcd4ae7f5af)

- مسیر فایل‌های سرتیفیکیت که در مرحله قبل دریافت کردید رو در جای خودشون قرار بدید.
- با حذف کاراکتر `#` از پشت خط `XRAY_SUBSCRIPTION_URL_PREFIX` اون رو از حالت کامنت خارج کنید. حالا سابدامنه‌ مربوط به لینک سابسکریپشن رو همراه با پورت پنل در این متغیر قرار بدید. دقت کنید آدرس لینک باید به صورت `https` باشه.
_(در صورت عدم مقداردهی به این متغیر، لینک سابسکریپشنی که از پنل دریافت میکنید با همون دامنه‌ای هست که وارد پنل شدید)_
تغییرات رو ذخیره کنید و در نهایت مرزبان رو ری‌استارت کنید:
```
marzban restart
```

حالا میتونید با دامنه `PANEL.DOMAIN1.COM` وارد پنل بشید. 
همچنین لینک سابسکریپشنی که از پنل دریافت میکنید با آدرس دامنه `SUB.DOMAIN2.COM` خواهد بود.
> توجه کنید که هم پنل و هم لینک سابسکریپشن با هر دو دامنه **قابل دسترسی** خواهند بود.

_نکنه: برای از کار نیفتادن لینک سابسکریپشن قبلی، باید طبق آموزش برای اون دامنه سرتیفیکیت بگیرید. میتونید برای سه یا چند دامنه هم سرتیفیکیت دریافت کنید._
_نکته: اگر از `HAproxy` استفاده میکنید، باید هر دو دامنه رو در کانفیگ `HAproxy` قرار بدید._

**Comment by happybigboy** [comment]


**Reply by mdjvd** [reply]
هر دو دامنه باید توی frontend مربوط به backend مرزبان قرار بگیرن:

```
 use_backend panel if { req.ssl_sni -m end PANEL.DOMAIN1.COM }
 use_backend panel if { req.ssl_sni -m end SUB.DOMAIN2.COM }

backend panel
 mode tcp
 server srv1 127.0.0.1:10000

**Reply by M03ED** [reply]
کانفیگ haproxy مقداری تغییرات نیاز داره 
روی لینک ساب به هر path به جز /sub دیتایی وارد بشه باید بلاک بشه

**Reply by mdjvd** [reply]
بله این درست‌تر و بهتره، ممنون. 
من ساده‌ ترین حالتشو گفتم که صرفا فقط کار کنه لینک ساب

**Reply by mahsamcse** [reply]
ممنون میشم بفرمایید تغیرات رو روی ha-proxy
اینطوری مطلب آموزشی این پست هم تکمیل تر میشه

**Comment by SahandSOne** [comment]
خیلی کاربردی بود. دمت گرم

**Reply by qirmizialma** [reply]
لطفا بفرمایید اگه بخواهیم از سرتیفیکیت های آماده ی کلادفلر استفاده کنیم چجوری باید انجام بدیم

**Reply by mdjvd** [reply]
فکر نمیکنم بشه. سرتیفیکیتی که کلودفلر issue میکنه فقط برای همون دامنه قابل استفاده هست تا جایی که من میدونم.

**Comment by rbsdotnet** [comment]
برای تک پورت کردن پنل و کتنفیگ ها شما چه روشی رو پیشنهاد می کنید  و اینکه ویدیو آموزشی هم هست معرفی کنید
باتشکر

**Reply by mdjvd** [reply]
https://gozargah.github.io/marzban/examples/all-on-one-port

**Reply by rbsdotnet** [reply]
در حالت VLESS TCP فالبک کار نمی کنه آیا شما راه حلی دارید؟

**Comment by silvermanq** [comment]
عالی بود برای من جواب داد.

**Comment by farshadl** [comment]
> سلام بر شما. ممکنه نیاز باشه دامنه پنل و دامنه لینک سابسکریپشن رو از همدیگه جدا کنید. (حالا به هر دلیلی) برای این کار چند روش وجود داره:
> 
> ۱- میتونید بعد از گرفتن یک سرتیفیکیت برای دامنه اول و فعالسازی SSL در مرزبان، فقط تیک پروکسی دامنه دوم رو فعال کنید و برای پنل یا لینک سابسکریپشن در نظر بگیریدش.
> 
> ۲- اگر میخواید از یک دامنه یکسان استفاده کنید ولی سابدامنه های متفاوت برای پنل و لینک سابسکریپشن بذارید، میتونید برای دامنه‌تون یک سرتیفیکیت `wildcard` بگیرید. (پیشنهاد نمیشه. بهتره از دامنه‌های کاملا متفاوت استفاده کنید.)
> 
> **۳- یک سرتیفیکیت برای دو دامنه بگیرید. توی این آموزش ما این کارو انجام میدیم.**
> 
> _خب برای گرفتن یک سرتیفیکیت برای دو دامنه، ما با `acme.sh` انجامش میدیم ولی شما میتونید از `certbot` هم استفاده کنید. (البته با دستورهای خودش)_
> 



الان ما دو تا سرتیفیکیت برای پنل و sub گرفتیم ولی فقط آدرس سرتیفیکیت پنل را وارد فایا env کردیم. کجا باید آدرس سرتیفیکیت sub را بدیم تا اطلاعات ساب‌دامنه sub هم رمزنگاری بشن؟؟؟؟؟؟

**Reply by mdjvd** [reply]
شما یک سرتیفیکیت برای دو دامنه میگیرید و در نتیجه فایل‌های فقط یک سرتیفیکیت رو به پنل معرفی میکنید، و این سرتیفیکیت برای هر دو دامنه valid هست و کار میکنه.

**Comment by Matialz7** [comment]
امکانش هست پنل و یا لینک ساب روی 2 پورت مجزا باشه ؟!!

**Reply by mdjvd** [reply]
به طور ساده نه. با ابزارهایی مثل nginx یا haproxy بله.

**Reply by yasinowo** [reply]
برای تمدیدش چیکار باید کرد؟

**Reply by mdjvd** [reply]
خودش باید خودکار تمدید بشه. ولی برای تمدید دستی از این دستور استفاده کنید. (دامنه اول توی دستوری که باهاش سرتیفیکیت گرفتید رو باید بذارید)
```bash
acme.sh --renew -d PANEL.DOMAIN1.COM --force
```

**Comment by shahdad1998** [comment]
درود وقت بخیر
من برای پنل قبلا ssl گرفتم. میخوام از همون دامنه (ساب دامنه) برای پنل و لینک ساب استفاده کنم که داره درست کار میکنه فعلا.
ولی میخوام برای یک دامنه جدید ssl بگیرم (یک ساب دامنه بر روی دامنه ای متفاوت از دامنه پنل و لینک ساب) که ازش برای کانفیگ های tls دار استفاده کنم.
امکانش هست؟ و چطور میتونم برای این ساب جدید privatekey و publickey رو بگیرم؟
چون توی مرزبان اینباند جنریتور که میخوام اینباند tls‌دار  بسازم این دوتا رو ازم میخواد. ممنون میشم راهنمایی کنید.

**Reply by mdjvd** [reply]
جداگونه برای دامنه جدیدتون سرتیفیکیت بگیرید و مسیرش رو توی اینباند بذارید. مرتبط به ssl پنل و ساب نیست.

**Comment by Mrkalier** [comment]
سلام کاربرا با هر دو دامنه میتونن کانفیگها ر اد کنن؟؟

**Reply by mdjvd** [reply]
سلام. متوجه منظورتون نشدم دقیق، کاربر که به هر دو دامنه دسترسی نداره خب. 
شما وقتی متغیر XRAY_SUBSCRIPTION_URL_PREFIX توی فایل .env رو مقداردهی میکنید و دامنه سابسکریپشن رو میذارید، لینک سابسکریپشنی که پنل براتون میسازه با این دامنه ای هست که اینجا گذاشتید. ولی اگه خودتون دستی لینک سابسکریپشن رو ادیت کنید و اون یکی دامنه رو به جاش بذارید با اون دامنه هم سابسکیرپشن کار میکنه براتون.

**Comment by alirezanbi** [comment]
اگر بخواهیم دامنه رو عوض کنیم باید چیکار کنیم؟

**Reply by mdjvd** [reply]
سرتیفیکیت جدید بگیرید و دامنه‌تون توی env رو عوض کنید

**Comment by arashmohebbati** [comment]
از کجا میتونیم مسیر path یا همون Dashboard رو تغییر بدیم؟
چون کسی که به سابسکریپشن دسترسی داشته باشه میتونه کلمه Dashboard اضافه کنه به پنل دسترسی پیدا کنه و با یه ddos بریم روی هوا

**Reply by mdjvd** [reply]
به طور ساده نمیشه هنوز، ولی با nginx میشه تغییرش داد

**Reply by mmsh1370** [reply]
میشه راه حل این کار رو بهمون بگین
هر کاری کردم با ngnix بتونم مسیر path یا همون Dashboard رو تغییر بدم اخرش حرف داشبورد تو آدرس اضافه میشه
من خیلی روش وقت گذاشتم و به نتیجه نرسیدم 
حس میکنم مرزبان از یه تغییر دهنده لینک داخل کد برنامه نویسی شده اش استفاده میکنه

**Reply by mdjvd** [reply]
خیلی ضرورت خاصی نداره که حتما path رو تغییر بدید. اگه یک یوزرنیم و پسورد قوی بذارید، اتفاقی قرار نیست بیفته.
احتمالا آپشن تغییر path هم بزودی به مرزبان اضافه میشه (یک PR باز هست براش)
باز اگه نیاز به راهنما داشتید، این دو نمونه شاید کمک کنن:

[1](https://github.com/fx505/Marzban-examples/blob/master/nginx-dashboard/nginx.conf)   [2](https://github.com/Gozargah/Marzban-examples/blob/21888ca1260610bc32cff37fe75bebcd4fde26b1/fully-single-port/nginx.conf)

**Comment by mehdikhody** [comment]
سلام. من برای پنل یک دامنه و برای لینک ساب یک دامنه متفاوت دارم. برای هر دوتا هم cert گرفتم اما تو تنظیمات جایی برای گذاشتن cert لینک ساب نیست و تو مرورگر میزنه نا امن.

**Reply by mdjvd** [reply]
سلام. مطلب رو بخونین لطفا. شما "یک" سرتیفیکیت برای "دو" دامنه میگیرین.

**Comment by dreamboo00st-a11y** [comment]
سلام 
سرتی که بهم داده به این صورته و به صورت pem نیست 

[Mon Aug  4 10:11:52 AM UTC 2025] Your cert is in: /root/.acme.sh/xxxmain.jingerpot.xyz_ecc/xxxxx.cer
[Mon Aug  4 10:11:52 AM UTC 2025] Your cert key is in: /root/.acme.shxxx_ecc/xxx.key
[Mon Aug  4 10:11:52 AM UTC 2025] The intermediate CA cert is in: /root/.acme.sh/xxxxx_ecc/ca.cer
[Mon Aug  4 10:11:52 AM UTC 2025] And the full-chain cert is in: /root/.acme.sh/xxxxx_ecc/fullchain.cer

جای اسم دامنه ها Xxx زدم که اینجا قابل انتشار باشه

**Reply by mdjvd** [reply]
مشکلی نداره، مراحل بعدی رو انجام بدید.

**Reply by davoodr22** [reply]
> مشکلی نداره، مراحل بعدی رو انجام بدید.

سلام و عرض ادب
بزرگوار من نیاز به یک سری بهینه سازی و تغیرات روی مرزبان دارم 
امکانش هست راه ارتباطی بدید؟
ایدی من تلگرام
davood_roz
